contracts/Bridge/XVSProxyOFTSrc.sol

```diff
@@ -20,13 +20,44 @@ contract XVSProxyOFTSrc is BaseXVSProxyOFT {
      */
     uint256 public outboundAmount;
 
+    /**
+     * @notice Emits when locked token released manually by owner.
+     */
+    event FallbackWithdraw(address indexed to, uint256 amount);
+    /**
+     * @notice Emits when stored message dropped without successfull retrying.
+     */
+    event DropFailedMessage(uint16 srcChainId, bytes srcAddress, uint64 nonce);
+
     constructor(
         address tokenAddress_,
         uint8 sharedDecimals_,
         address lzEndpoint_,
         address oracle_
     ) BaseXVSProxyOFT(tokenAddress_, sharedDecimals_, lzEndpoint_, oracle_) {}
 
+    /** @notice Only call it when there is no way to recover the failed message.
+     * `dropFailedMessage` must be called first if transaction is from remote->local chain to avoid double spending.
+     * @param to_ The address to withdraw to
+     * @param amount_ The amount of withdrawal
+     */
+    function fallbackWithdraw(address to_, uint256 amount_) external onlyOwner {
+        outboundAmount -= amount_;
+        _transferFrom(address(this), to_, amount_);
+        emit FallbackWithdraw(to_, amount_);
+    }
+
+    /** @notice Only call it when there is no way to recover the failed message.
+     * `dropFailedMessage` must be called first to avoid double spending.
+     * @param srcChainId_ Chain id of source
+     * @param srcAddress_ Address of source followed by current bridge address
+     * @param nonce_ Nonce_ of the transaction
+     */
+    function dropFailedMessage(uint16 srcChainId_, bytes memory srcAddress_, uint64 nonce_) external onlyOwner {
+        failedMessages[srcChainId_][srcAddress_][nonce_] = bytes32(0);
```

Comment:
Would it make sense to decode the message before removing it, and transfer the needed funds from this function (if needed). This way we could remove the function `fallbackWithdraw`, avoiding potential errors invoking that function without dropping the message first

Reply:
Case 1 :  `failedMessages[srcChainId_][srcAddress_][nonce_]` will be updated when tokens coming from `dest->src` and it failed, so in order to return the token, we must first clear this stored message and then use fallbackWithdraw.

Case 2 : When user send tokens from src->dest and in any case of we are unable to retry msg on destination we can directly use fallbackWithdraw to return the tokens.